src/urllib3/contrib/pyopenssl.py

```diff
@@ -53,13 +53,22 @@ class UnsupportedExtension(Exception):  # type: ignore[no-redef]
 
 import logging
 import ssl
+import warnings
 from io import BytesIO
 from socket import socket as socket_cls
 from socket import timeout
 from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union
 
 from .. import util
 
+warnings.warn(
```

Comment:
You might want this warning only when you construct PyOpenSSLContext or call `inject_into...`